code/core/write_sdl_ntuple.cc

```diff
@@ -459,88 +323,39 @@ std::tuple<float, float, float, vector<unsigned int>, vector<unsigned int>> pars
     const float pt = (ptAv_in + ptAv_out) / 2.;
 
     // T5 eta and phi are computed using outer and innermost hits
-    SDL::CPU::Hit hitA(trk.ph2_x()[i1_idx], trk.ph2_y()[i1_idx], trk.ph2_z()[i1_idx]);
-    SDL::CPU::Hit hitB(trk.ph2_x()[i5_idx], trk.ph2_y()[i5_idx], trk.ph2_z()[i5_idx]);
+    SDL::CPU::Hit hitA(trk.ph2_x()[Hit_0], trk.ph2_y()[Hit_0], trk.ph2_z()[Hit_0]);
+    SDL::CPU::Hit hitB(trk.ph2_x()[Hit_8], trk.ph2_y()[Hit_8], trk.ph2_z()[Hit_8]);
```

Comment:
The comment says that we are to compute η based on the outermost hit. But doesn't a T5 have 10 hits always? If so, why do we take the 9th hit? Do we need the lower one from the outer MD? If so, I think this is again an inconsistency wrt. how we take hits for pTXs...

Reply:
I believe it is possible that a T5 be made out of 5 PS modules, (if it's very forward for example.) in which case, the position of the last hit is best known via a pixel hit, instead of the strip hit.
And the ordering of the hits is made always such taht the 9th hit is the "anchor hit" i.e. always a pixel hit if the last two hits are PS module.
Hence, I picked 9th hit which is the anchor hit.

in other words, we always have the hits in the following order.
1. anchor hit
2. other hit
3. anchor hit
4. other hit
...
...
9. anchor hit
10. other hit

And so I chose the "anchor hit" of the last mini-doublet to represent the hit position.